app/src/main/java/com/example/plantsapp/presentation/ui/plants/PlantsFragment.kt

```diff
@@ -32,8 +32,10 @@ class PlantsFragment : Fragment(R.layout.fragment_plants) {
             }
 
             clickedPlant.observe(viewLifecycleOwner) {
-                it.getContentIfNotHandled()?.let {
-                    openFragment(PlantDetailFragment())
+                it.getContentIfNotHandled()?.let { plant ->
+                    openFragment(
+                        PlantDetailFragment.newInstance(plant.name)
```

Comment:
Here you can use value class instead of just String 
![image](https://user-images.githubusercontent.com/23121846/137927815-bce6d9f0-36f0-4848-b05b-c77ce1a8bd1f.png)